src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -397,26 +404,52 @@ public int getPassCount() {
      */
     @Exported(visibility=9)
     public int getFailedSince() {
-        // If we haven't calculated failedSince yet, and we should,
-        // do it now.
+        // If we haven't calculated failedSince yet, and we should, do it now.
         if (failedSince==0 && getFailCount()==1) {
             CaseResult prev = getPreviousResult();
-            if(prev!=null && !prev.isPassed())
+            if(prev!=null && !prev.isPassed()) {
                 this.failedSince = prev.getFailedSince();
+            }
             else if (getRun() != null) {
                 this.failedSince = getRun().getNumber();
             } else {
                 LOGGER.warning("trouble calculating getFailedSince. We've got prev, but no owner.");
-                // failedSince will be 0, which isn't correct. 
+                // failedSince will be 0, which isn't correct.
             }
         }
         return failedSince;
     }
-    
+
+    /**
+     * If this test was skipped, then return the build number
+     * when this test was first skipped.
+     */
+    @Exported(visibility=9)
+    public int getSkippedSince() {
+        // If we haven't calculated skippedSince yet, and we should, do it now.
+        if (skippedSince==0 && getSkipCount()==1) {
+            CaseResult prev = getPreviousResult();
+            if(prev!=null && prev.isSkipped()){
+                this.skippedSince = prev.getSkippedSince();
```

Comment:
Unlike failedSince, skippedSince is not available in XML, so this could recursively load all the skipped builds leading to StackOverflowError like [JENKINS-31660](https://issues.jenkins-ci.org/browse/JENKINS-31660) . IMHO it would be safer to save skippedSince in XML too and assume current build is the first skipped one if no information is found in XML, avoiding recursion. (Related to #117.)

Reply:
Good catch! I will review this and suggest a fix

Reply:
I've given this some thought.. as I understand we only parse the XML and have no control over its contents. Do you see any other way to protect against a StackOverflowError ?